.github/workflows/script/solo_smoke_test.sh

```diff
@@ -0,0 +1,228 @@
+
+  LOCAL_NODE_KEYS=$(cat private_key_with_quote_final.txt)
+
+  echo "Add new keys to hardhat.config.js"
+  git checkout test/smoke/hardhat.config.js
+  awk '/accounts: \[/ {print; getline; getline; next} 1' test/smoke/hardhat.config.js > test/smoke/hardhat.config.js.tmp
+
+  if [[ "$OSTYPE" == "darwin"* ]]; then
+    echo "$LOCAL_NODE_KEYS" > temp.txt
+    sed '/accounts: \[/r temp.txt'  test/smoke/hardhat.config.js.tmp  > test/smoke/hardhat.config.js
+    rm temp.txt
+  else
+    awk -v new_keys="$LOCAL_NODE_KEYS" '/accounts: \[/ {print; print new_keys; next} 1' test/smoke/hardhat.config.js.tmp > test/smoke/hardhat.config.js || true
+  fi
+  echo "Display the new hardhat.config.js"
+  cat test/smoke/hardhat.config.js
+
+  if [ -s "private_key_without_quote_final.txt" ]; then
+    echo "File private_key_without_quote_final.txt exists and not empty"
+    return 0
+  else
+    echo "File private_key_without_quote_final.txt does not exist or empty"
+    return 1
+  fi
+}
+
+function clone_smart_contract_repo ()
+{
+  echo "Clone hedera-smart-contracts"
+  if [ -d "hedera-smart-contracts" ]; then
+    echo "Directory hedera-smart-contracts exists."
+  else
+    echo "Directory hedera-smart-contracts does not exist."
+    git clone https://github.com/hashgraph/hedera-smart-contracts --branch only-erc20-tests
+  fi
+}
+
+function clone_sdk_repo ()
+{
+  echo "Clone hedera-sdk-js"
+  if [ -d "hedera-sdk-js" ]; then
+    echo "Directory hedera-sdk-js exists."
+  else
+    echo "Directory hedera-sdk-js does not exist."
+    git clone https://github.com/hashgraph/hedera-sdk-js --branch v2.53.0
+    cd hedera-sdk-js
+    npm install --save @hashgraph/sdk
+    cd -
+  fi
+}
+
+# clone hedera local node repo and call function create_account_and_extract_key
+# to extract private keys
+function clone_local_node_repo ()
+{
+  echo "Clone hedera local node"
+  if [ -d "hedera-local-node" ]; then
+    echo "Directory hedera-local-node exists."
+  else
+    echo "Directory hedera-local-node does not exist."
+    git clone https://github.com/hashgraph/hedera-local-node --branch v2.32.0
+  fi
+  cd hedera-local-node
+  npm install
+
+  function_name=create_account_and_extract_key
+  retry_function 5
+  cd -
+}
+
+# retry the function saved in function_name for few times
+function retry_function ()
+{
+  local num=$1
+  for ((i=1; i<=$num; i++)); do
+    return_code=$( ( $function_name > retry.log; echo $? ))
+    # Use the return code
+    if [[ $return_code -eq 0 ]]; then
+      echo "Function $function_name is successful"
+      return
+    else
+      echo "Function $function_name failed with return code $return_code"
+    fi
+    echo "Retry $function_name in 2 seconds"
+    sleep 2
+  done
+  echo "Function $function_name failed after 5 retries"
+  exit 1
+}
+
+function setup_smart_contract_test ()
+{
+  echo "Setup smart contract test"
+  CONTRACT_TEST_KEYS=$(cat hedera-local-node/private_key_without_quote_final.txt)
+
+  cd hedera-smart-contracts
```

Comment:
just get a copy of the smart contracts we want and check them in our repo in a good location